code/script.js

```diff
@@ -262,82 +270,148 @@ const selectQuestion = () => {
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
 
+  const questionValue = questions.value
+
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
-      category: category,
+      value: questionValue,
+      category: category
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: questionValue,
+      category: category
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
-      // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
-      value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
-      category: category,
+      attribute: questionValue,
+      value: true,
+      category: category
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: questionValue,
+      value: true,
+      category: category
+    }
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
+  selectQuestion()
   // Compare the currentQuestion with the secret person.
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
+  guesses++
+  guessCounter.innerHTML = `<p>Number of guesses: ${guesses}</p>`
+  if (currentQuestion.value === secret[currentQuestion.attribute]) {
+    keep = true
+  } else {
+    keep = false
+  }
+  filterCharacters(keep)
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
```

Comment:
It's easy to understand where all code (group, attribute, value) is coming from and how it fits in the filter function.